core/src/main/java/com/scalar/db/storage/jdbc/JdbcDatabaseAdmin.java

```diff
@@ -76,4 +190,174 @@ public void close() {
+          if (e.getErrorCode() != 1920) {
+            throw e;
+          }
+        }
+        execute(
+            connection,
+            "ALTER USER "
+                + enclose(getFullNamespaceName(schemaPrefix, schema))
+                + " quota unlimited on USERS");
+        break;
+    }
+  }
+
+  @SuppressFBWarnings("SQL_NONCONSTANT_STRING_PASSED_TO_EXECUTE")
+  private void createTableInternal(
+      Connection connection, String schema, String table, TableMetadata metadata)
+      throws SQLException {
+    String createTableStatement =
+        "CREATE TABLE "
+            + enclosedFullTableName(getFullNamespaceName(schemaPrefix, schema), table, rdbEngine)
+            + "(";
+    LinkedHashSet<String> sortedColumnNames =
+        Sets.newLinkedHashSet(
+            Iterables.concat(
+                metadata.getPartitionKeyNames(),
+                metadata.getClusteringKeyNames(),
+                metadata.getColumnNames()));
+    // Add columns definition
+    createTableStatement +=
+        sortedColumnNames.stream()
+            .map(
+                columnName ->
+                    enclose(columnName) + " " + getVendorDbColumnType(metadata, columnName))
+            .collect(Collectors.joining(","));
+    // Add primary key definition
+    createTableStatement +=
+        ", PRIMARY KEY ("
+            + Stream.concat(
+                    metadata.getPartitionKeyNames().stream(),
+                    metadata.getClusteringKeyNames().stream())
+                .map(this::enclose)
+                .collect(Collectors.joining(","))
+            + "))";
+    execute(connection, createTableStatement);
+  }
+
+  /**
+   * Get the vendor DB data type that is equivalent to the Scalar DB data type
+   *
+   * @param metadata a table metadata
+   * @param columnName a column name
+   * @return a vendor DB data type
+   */
+  private String getVendorDbColumnType(TableMetadata metadata, String columnName) {
+    HashSet<String> keysAndIndexes =
+        Sets.newHashSet(
+            Iterables.concat(
+                metadata.getPartitionKeyNames(),
+                metadata.getClusteringKeyNames(),
+                metadata.getSecondaryIndexNames()));
+    DataType scalarDbColumnType = metadata.getColumnDataType(columnName);
+    if (keysAndIndexes.contains(columnName)) {
+      return DATA_TYPE_MAPPING_FOR_KEY
+          .get(rdbEngine)
+          .getOrDefault(
+              scalarDbColumnType, DATA_TYPE_MAPPING.get(rdbEngine).get(scalarDbColumnType));
+    } else {
+      return DATA_TYPE_MAPPING.get(rdbEngine).get(scalarDbColumnType);
+    }
+  }
+
+  private void createIndex(
+      Connection connection, String schema, String table, TableMetadata metadata)
+      throws SQLException {
+    LinkedHashSet<String> sortedIndexedColumns = new LinkedHashSet<>(metadata.getColumnNames());
+    sortedIndexedColumns.retainAll(metadata.getSecondaryIndexNames());
```

Comment:
Right, sorry about that. I wanted to keep the index creation sorted with the same order it was declared when creating the `TableMetadata` object but sorting it after the `columns` do not make sense. 🤦 
Is there any reason that `TableMetadata.secondaryIndexNames` is backed by a `HashSet` and not a `LinkedHashSet` to preserve the index declaration order ?

Reply:
> Is there any reason that TableMetadata.secondaryIndexNames is backed by a HashSet and not a LinkedHashSet to preserve the index declaration order?

This is because we don't need to preserve the index declaration order as we only support single column secondary indexes in Scalar DB for now. So we can create the declared indexes in any order.